_posts/2020-01-01-2019-summary.md

```diff
@@ -0,0 +1,230 @@
+let’s take a closer look at everything that went in. And there is a lot to look at!
+
+<!--more-->
+
+# Improvements
+
+## Language Features
+
+* [`FETCH FIRST n ROWS [ONLY | WITH TIES]`](/docs/current/sql/select.html#limit-or-fetch-first-clauses) 
+  standard syntax. The `WITH TIES` clause is particularly useful when some of the rows have the same 
+  value for the columns being used to order the results of a query. Consider a case where you want to 
+  list top 5 students with highest score on an exam. if 6th person has the same score as the 5th, you 
+  want to know this as well, instead of getting an arbitrary and non-deterministic result:
+
+  ```sql
+  SELECT student_name, score
+  FROM student s JOIN exam_result e USING (student_id)
+  ORDER BY score
+  FETCH FIRST 5 ROWS WITH TIES
+  ```
+
+* [`OFFSET`](/docs/current/sql/select.html#offset-clause) syntax, which is especially useful in ad-hoc queries.
+* [`COMMENT ON <table>`](/docs/current/sql/comment.html) syntax to 
+  set or remove table comments, and ability to see table comments via `DESCRIBE` or the 
+  `system.metadata.table_comments` table.
+* Support for ``LATERAL`` in the context of an outer join.
+* Support for ``UNNEST`` in the context of ``LEFT JOIN``. With this feature, it is now possible 
+  to preserve the outer row when the array contains zero elements or is NULL. Most common usages
+  of `UNNEST` in a `CROSS JOIN` should actually be using this form.
+	
+  ```sql	
+  SELECT * FROM t LEFT JOIN UNNEST(t.a) u(v) ON true
+  ```
+
+* Window functions now support the IGNORE NULLS clause. This is useful when combined with 
+  functions such as LEAD, LAG, FIRST_VALUE, LAST_VALUE and NTH_VALUE if the dataset contains nulls.
+* `ROW` expansion using `.*` operator.
+* `ROW` types now support the subscript operator to access fields by index. This This greatly improves usability 
+  and readability of queries when dealing with ROW type containing anonymous fields.
+  
+    ![](/assets/blog/2019-review/row-ordinal.png)
+
+* [`CREATE SCHEMA`](/docs/current/sql/create-schema.html) syntax and support 
+  in various connectors (Hive, Iceberg, MySQL, PostgreSQL, Redshift, SQL Server, Phoenix).
+* Support for correlated subqueries containing `LIMIT` or `ORDER BY`+`LIMIT`.
+
+## Query Engine
+
+* Generalize conditional, lazy loading and processing (a.k.a., late materialization) beyond 
+  Table Scan, Filter and Projection to support Join, Window, TopN and Semijoin operators. This can dramatically 
+  reduce latency, CPU and I/O for highly selective queries. This is one of the most important performance 
+  optimizations in recent times and we will be blogging about this more in coming weeks.
+* [Unwrap cast/predicate pushdown]({% post_url 2019-05-21-optimizing-the-casts-away %}) optimizations.
+* Add support for pushing down operations such as limit, table sample, or projections into connectors during
```

Comment:
We should avoid "add support for" here since this is a list of features (nouns).

> Connector pushdown during planning for operations such as limit, table sample, or projections.

Reply:
I don't see what's wrong with that form, but I'll change it.